crates/red_knot_python_semantic/resources/mdtest/binary/booleans.md

```diff
@@ -0,0 +1,25 @@
+## Binary operations on booleans
+
+## Basic Arithmetic
+
+```py
+a = True
+b = True
+c = a + b
+d = a - b
+e = a * b
+f = a % b
+g = a // b
+h = a ** b
+i = a / b
+j = a | b
+
+reveal_type(c)  # revealed: int
+reveal_type(d)  # revealed: int
+reveal_type(e)  # revealed: int
+reveal_type(f)  # revealed: int
+reveal_type(g)  # revealed: int
+reveal_type(h)  # revealed: int
+reveal_type(i)  # revealed: float
+reveal_type(j)  # revealed: bool
```

Comment:
I think we can be more precise in all of these cases (except for division, since we don't have float literals). For example, we know that `True + True == 2`, so that can be `Literal[2]` instead of `int`. In fact I think the best implementation strategy for all these cases (except for `|`, where we can return a `BooleanLiteral`) is to implement the operator for `IntLiteral` and then just have binary operations on `BooleanLiteral` call back in to the same operation, but with `BooleanLiteral[True]` converted to `Literal[1]` and `BooleanLiteral[False]` converted to `Literal[0]`.